### PR TITLE
Integrate wm collapsible list item

### PR DESCRIPTION
**Pull request by dvira**
- Refactored `CourseOutlineLessonItem` to use `WMCollapsibleListItem`.
- Updated its delete dialog.
- Updated its empty state.
- Added optimizations.

TODOs:
- Refine behavior and styles when dragging an item over empty lesson.
- Improve DnD behavior and styles.

### CODE REVIEW

File: src/components/Screen-v2/CourseEditorScreen/CourseOutline/CourseOutlineTab/CourseItemsList/index.tsx [hunk 1]
```diff
@@ -32,22 +32,25 @@ export default function CourseItemsList({
...
+
+  const [isDragging, setIsDragging] = useState(false);
 
   return (
     <div className={cc([classes['course-items-list'], className])}>
       <Container
         {...otherProps}
         getChildPayload={(i) => items[i]}
+        onDragEnter={() => setIsDragging(true)}
+        onDragLeave={() => setIsDragging(false)}
```
Comment (dr15) [inline src/components/Screen-v2/CourseEditorScreen/CourseOutline/CourseOutlineTab/CourseItemsList/index.tsx:49]: Consider wrapping these in `useCallback()`

File: src/components/Screen-v2/CourseEditorScreen/CourseOutline/CourseOutlineTab/CourseOutlineLessonItem/index.tsx [hunk 2]
```diff
@@ -1,121 +1,161 @@
...
+            fontSize={14}
+          />
         }
-        hasDragHandle
-        ref={lessonRef}
+        isNew={item.id === newLessonId && isEmpty}
+        extraActions={[
+          {
+            iconComponent: <WMIconDelete />,
+            onClick: () => (isEmpty ? deleteLesson() : setOpenDialog(true)),
```
Comment (dr15) [inline src/components/Screen-v2/CourseEditorScreen/CourseOutline/CourseOutlineTab/CourseOutlineLessonItem/index.tsx:122]: Consider wrapping this in `useCallback()`

File: src/components/Screen-v2/CourseEditorScreen/CourseOutline/CourseOutlineTab/CourseOutlineLessonItem/index.tsx [hunk 3]
```diff
@@ -1,121 +1,161 @@
...
-          isActive={(item: any) => item.id === activeDetailsItem?.id}
+          isActive={checkIsActive}
         />
-      </WMCollapse>
-    </Draggable>
+      </WMCollapsibleListItem>
+      <WMConfirmationDialog
+        open={openDialog}
+        onClose={() => setOpenDialog(false)}
+        onCancel={() => setOpenDialog(false)}
```
Comment (dr15) [inline src/components/Screen-v2/CourseEditorScreen/CourseOutline/CourseOutlineTab/CourseOutlineLessonItem/index.tsx:147]: Consider wrapping this in `useCallback()`

**Review by dr15 — APPROVED** [review]
Approved but left a few comments.

I noticed you wrote that design and behavior need to be improved. Behavior is a bit more complicated to improve so I won't comment about it. Here are some style flaws I found that you might be able to fix now?

Weird gaps:




Missing white background:

File: src/components/Screen-v2/CourseEditorScreen/CourseOutline/CourseOutlineTab/CourseOutlineLessonItem/index.tsx (hunk 3 above)
Comment (dvira) [reply src/components/Screen-v2/CourseEditorScreen/CourseOutline/CourseOutlineTab/CourseOutlineLessonItem/index.tsx:147]: Done

File: src/components/Screen-v2/CourseEditorScreen/CourseOutline/CourseOutlineTab/CourseOutlineLessonItem/index.tsx (hunk 2 above)
Comment (dvira) [reply src/components/Screen-v2/CourseEditorScreen/CourseOutline/CourseOutlineTab/CourseOutlineLessonItem/index.tsx:122]: Done

File: src/components/Screen-v2/CourseEditorScreen/CourseOutline/CourseOutlineTab/CourseItemsList/index.tsx (hunk 1 above)
Comment (dvira) [reply src/components/Screen-v2/CourseEditorScreen/CourseOutline/CourseOutlineTab/CourseItemsList/index.tsx:49]: Done

**Comment by dvira** [comment]
Those weird gaps and the list item's transparent background should be better handled once the new `WMListItem` is fully integrated.
So, I left a todo to fix these discrepancies. Thanks for your input!

**Pull request merged**